### ISSUE TITLE
Initialization code that relies on evolutions fails with exceptions

### ISSUE THREAD

**Issue by denisftw** [playframework/playframework#9028]
### Description

Initialization code that runs immediately after evolutions and relies on them being applied causes exceptions in Play 2.7, but works well in Play 2.6.20.

### Play Version

2.7.0

### API

Scala

### Operating System

Linux dell 4.4.0-142-generic #168-Ubuntu SMP Wed Jan 16 21:00:45 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

### JDK

openjdk version "1.8.0-adoptopenjdk"
OpenJDK Runtime Environment (build 1.8.0-adoptopenjdk-jenkins_2018_05_19_01_00-b00)
OpenJDK 64-Bit Server VM (build 25.71-b00, mixed mode)

### Expected Behavior

I'm using compile-time dependency injection and have a block called [`start`](https://github.com/denisftw/play-evolutions-test/blob/master/app/AppLoader.scala#L53). In this block I initialize the database, evaluate `applicationEvolutions` (which is supposed create the `users` table) and then query the table that was just created. In theory, it should work because the querying code runs after evolutions, and in fact, in 2.6.20 everything works as expected.

### Actual Behavior

An exception from the DB engine complaining that the table doesn't exist. I'm using H2 for demo, but the same thing happens with PostgreSQL. As far as I can tell, evolutions are eventually applied but not when expected.

### Reproducible Test Case

The [`master`](https://github.com/denisftw/play-evolutions-test/tree/master) contains code that uses Play 2.7.0 and fails; the branch [`play26`](https://github.com/denisftw/play-evolutions-test/tree/play26) uses 2.6.20 and works. The only difference is Play version switch:
https://github.com/denisftw/play-evolutions-test/compare/play26

**Comment by mkurz** [comment]
This is documented in the migration guide:
https://www.playframework.com/documentation/2.7.x/Migration27#Application-starts-when-evolutions-scripts-need-to-be-applied-in-DEV-mode

What you need to do is to check if the database is up-to-date:
```scala
if (applicationEvolutions.upToDate) {
   // do the stuff that relies on evolutions here
}
```

**Comment by denisftw** [comment]
@mkurz Yeah, you're right! I guess I needed to pay more attention to migration notes ><

Now I checked [this discussion](https://github.com/playframework/playframework/issues/8935), but I'm not convinced that the flag is the best solution. If I add this condition, then I'll avoid the exception, but the code will never be executed either. 

I agree that making evolutions more user-friendly is a good cause, but not being able to rely on updated schema is very inconvenient. I think this issue should be closed, but ideally the solution needs to be more reactive (like `Future` or callback).
